packages/ui/src/ui-component/extended/TextToSpeech.jsx

```diff
@@ -0,0 +1,624 @@
+
+    const [textToSpeech, setTextToSpeech] = useState(null)
+    const [selectedProvider, setSelectedProvider] = useState('none')
+    const [voices, setVoices] = useState([])
+    const [loadingVoices, setLoadingVoices] = useState(false)
+
+    const onSave = async () => {
+        const textToSpeechConfig = setValue(true, selectedProvider, 'status')
+        try {
+            const saveResp = await chatflowsApi.updateChatflow(dialogProps.chatflow.id, {
+                textToSpeech: JSON.stringify(textToSpeechConfig)
+            })
+            if (saveResp.data) {
+                enqueueSnackbar({
+                    message: 'Text To Speech Configuration Saved',
+                    options: {
+                        key: Date.now() + Math.random(),
+                        variant: 'success',
+                        action: (key) => (
+                            <Button style={{ color: 'white' }} onClick={() => closeSnackbar(key)}>
+                                <IconX />
+                            </Button>
+                        )
+                    }
+                })
+                dispatch({ type: SET_CHATFLOW, chatflow: saveResp.data })
+            }
+        } catch (error) {
+            enqueueSnackbar({
+                message: `Failed to save Text To Speech Configuration: ${
+                    typeof error.response.data === 'object' ? error.response.data.message : error.response.data
+                }`,
+                options: {
+                    key: Date.now() + Math.random(),
+                    variant: 'error',
+                    persist: true,
+                    action: (key) => (
+                        <Button style={{ color: 'white' }} onClick={() => closeSnackbar(key)}>
+                            <IconX />
+                        </Button>
+                    )
+                }
+            })
+        }
+    }
+
+    const setValue = (value, providerName, inputParamName) => {
+        let newVal = {}
+        if (!textToSpeech || !Object.hasOwn(textToSpeech, providerName)) {
+            newVal = { ...(textToSpeech || {}), [providerName]: {} }
+        } else {
+            newVal = { ...textToSpeech }
+        }
+
+        newVal[providerName][inputParamName] = value
+        if (inputParamName === 'status' && value === true) {
+            // ensure that the others are turned off
+            Object.keys(textToSpeechProviders).forEach((key) => {
+                const provider = textToSpeechProviders[key]
+                if (provider.name !== providerName) {
+                    newVal[provider.name] = { ...(textToSpeech?.[provider.name] || {}), status: false }
+                }
+            })
+            if (providerName !== 'none' && newVal['none']) {
+                newVal['none'].status = false
+            }
+        }
+        setTextToSpeech(newVal)
+        return newVal
+    }
+
+    const handleProviderChange = (provider, configOverride = null) => {
```

Comment:
after selecting, the popup doesnt close:

<img width="1442" height="276" alt="image" src="https://github.com/user-attachments/assets/c084781e-3b16-4286-ac16-56397752f793" />